### PR TITLE
Development

### PR DESCRIPTION
**Pull request by alimoncul**

### CODE REVIEW

File: .ui/library/flAlert.cpx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (alimoncul) [inline .ui/library/flAlert.cpx:1]: we switched to native alert iirc. Could you remove this?

File: .ui/library/flDialog.cpx [hunk 2]
```diff
@@ -0,0 +1,106 @@
+{
```
Comment (alimoncul) [inline .ui/library/flDialog.cpx]: we switched to native alert iirc. Could you remove this?

File: .ui/library/gviHomeCategoryItem.cpx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+			},
+			"type": "GridViewItem",
+			"userProps": {
+				"flexProps": {},
+				"left": "23.40",
+				"top": "30.85"
+			}
+		},
+		{
+			"className": ".sf-flexLayout .gviHomeCategoryItem-flHomeCategoryWrapper",
```
Comment (alimoncul) [inline .ui/library/gviHomeCategoryItem.cpx]: this should have been a separate component.

File: .ui/library/lviEmptyItem.cpx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+			"userProps": {
+				"flexProps": {
+					"flexGrow": null
+				},
+				"height": 150,
+				"usePageVariable": true
+			}
+		},
+		{
+			"className": ".sf-label .lviEmptyItem-flWrapper-lblEmptyTitle",
```
Comment (alimoncul) [inline .ui/library/lviEmptyItem.cpx]: never use font styles inside a component class. Font styles should be used inside a text style file.

File: .ui/library/lviHomeCategories.cpx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+				"parent": "57f4-201f-4bfc-5fc6"
+			},
+			"source": {
+				"page": "__library__"
+			},
+			"type": "ListViewItem",
+			"userProps": {}
+		},
+		{
+			"className": ".sf-flexLayout",
```
Comment (alimoncul) [inline .ui/library/lviHomeCategories.cpx]: Why this flex exists? We should remove it if no special styles used in it.

File: .ui/library/lviHomeCategories.cpx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+				],
+				"name": "gvCategories",
+				"parent": "0595-307b-ea39-8bec",
+				"usePageVariable": true
+			},
+			"source": {
+				"page": "__library__"
+			},
+			"type": "GridView",
+			"userProps": {
```
Comment (alimoncul) [inline .ui/library/lviHomeCategories.cpx:35]: these props should be inside a class of component.

File: .ui/library/lviHomeProducts.cpx [hunk 7]
```diff
@@ -16,16 +16,17 @@
...
-				"left": null,
-				"top": null
+				"flexProps": {},
+				"left": "-3.57",
+				"top": "333.93"
 			}
 		},
 		{
-			"className": ".sf-flexLayout .flProductContainer",
+			"className": ".sf-flexLayout .flProductContainer .grow",
```
Comment (alimoncul) [inline .ui/library/lviHomeProducts.cpx]: Empty class (flProductContainer)

File: .ui/library/lviProfile.cpx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+			},
+			"type": "ListViewItem",
+			"userProps": {
+				"flexProps": {},
+				"left": "0",
+				"top": "118.97"
+			}
+		},
+		{
+			"className": ".sf-flexLayout .flAccountUser",
```
Comment (alimoncul) [inline .ui/library/lviProfile.cpx:25]: "type": "FlexLayout" but uses a component class. If its a component use the component not the component's class only.

File: .ui/library/lviShowcaseHeader.cpx [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+				"parent": "57f4-201f-4bfc-5fc6"
+			},
+			"source": {
+				"page": "__library__"
+			},
+			"type": "ListViewItem",
+			"userProps": {}
+		},
+		{
+			"className": ".sf-flexLayout .flProductContainer-flProductHeader",
```
Comment (alimoncul) [inline .ui/library/lviShowcaseHeader.cpx:21]: "type": "FlexLayout" but uses a component class

File: .ui/library/lviSpacer.cpx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+			"source": {
+				"page": "__library__"
+			},
+			"type": "ListViewItem",
+			"userProps": {
+				"flexProps": {}
+			}
+		},
+		{
+			"className": ".sf-flexLayout .lviSpacer-dummyLayout .grow-relative",
```
Comment (alimoncul) [inline .ui/library/lviSpacer.cpx]: If its just empty lvi we shouldn't using a separate layout inside. Just assign colors in .lviSpacer

File: .ui/pgAccount.pgx [hunk 11]
```diff
@@ -54,53 +54,8 @@
 				},
 				"title": "pgAccount",
 				"visible": true
```
Comment (alimoncul) [inline .ui/pgAccount.pgx]: we shouldn't using userprops inside a page

File: .ui/pgCart.pgx [hunk 12]
```diff
@@ -47,49 +50,46 @@
 				"backgroundColor": "rgba( 255, 255, 255, 1 )",
```
Comment (alimoncul) [inline .ui/pgCart.pgx]: we shouldn't be using user props inside page

File: .ui/pgCategories.pgx [hunk 13]
```diff
@@ -87,18 +67,20 @@
 				"layoutManager": {
 					"spanCount": 2
```
Comment (alimoncul) [inline .ui/pgCategories.pgx]: we shouldn't be using user props inside page

File: .ui/pgCategoryDetail.pgx [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+			"props": {
+				"children": [],
+				"isRemovable": false,
+				"name": "headerBar",
+				"parent": "7dd3-eb16-f5e0-3035"
+			},
+			"type": "HeaderBar",
+			"userProps": {
+				"backgroundColor": "rgba( 255, 255, 255, 1 )",
+				"visible": true
```
Comment (alimoncul) [inline .ui/pgCategoryDetail.pgx]: we shouldn't be using user props inside page

File: .ui/pgFavorites.pgx [hunk 15]
```diff
@@ -50,10 +49,8 @@
 			"className": ".sf-listView .grow",
```
Comment (alimoncul) [inline .ui/pgFavorites.pgx]: we shouldn't be using user props inside page (for headerbar)

File: .ui/pgHome.pgx [hunk 16]
```diff
@@ -87,43 +68,6 @@
 				"testId": "Z--GuhP9L",
```
Comment (alimoncul) [inline .ui/pgHome.pgx]: we shouldn't be using user props inside page

File: .ui/pgHomeSlider.pgx [hunk 17]
```diff
@@ -60,6 +60,7 @@
 					"flexGrow": 1
 				},
 				"height": null,
+				"imageFillType": "ASPECTFIT",
```
Comment (alimoncul) [inline .ui/pgHomeSlider.pgx]: we shouldn't be using user props inside page

File: scripts/components/FlProductItem.ts [hunk 18]
```diff
@@ -1,10 +1,98 @@
...
-	}
+    pageName?: string | undefined;
+    myActivityIndicator: ActivityIndicator;
+    constructor(props?: any, pageName?: string) {
+        // Initalizes super class for this scope
+        super(props);
+        this.pageName = pageName;
+    }
+    initIndicator() {
+        this.myActivityIndicator = new ActivityIndicator();
```
Comment (alimoncul) [inline scripts/components/FlProductItem.ts]: add indicator in cpx not from code

File: scripts/components/FlProductItem.ts [hunk 19]
```diff
@@ -1,10 +1,98 @@
...
+            right: 13,
+            visible: true,
+            flexProps: {
+                positionType: 'ABSOLUTE'
+            },
+            color: '#181725'
+        });
+    }
+    toggleIndicator(toggle: boolean): void {
+        console.log('toggle', toggle);
```
Comment (alimoncul) [inline scripts/components/FlProductItem.ts]: remove log

File: scripts/components/FlProductItem.ts [hunk 20]
```diff
@@ -1,10 +1,98 @@
...
+            userStyle: {
+                visible: toggle
+            }
+        });
+    }
+    get onActionClick(): Button['onTouch'] {
+        return this.btnAddToBasket.onTouch;
+    }
+    set onActionClick(value: Button['onTouch']) {
+        this.btnAddToBasket.onTouch = value;
```
Comment (alimoncul) [inline scripts/components/FlProductItem.ts]: use event emitter events

File: scripts/components/FlProductItem.ts [hunk 21]
```diff
@@ -1,10 +1,98 @@
...
+    }
+    get itemTitle(): string {
+        return this.lblProductItemTitle.text;
+    }
+    set itemTitle(value: string) {
+        this.lblProductItemTitle.text = value;
+    }
+    get itemPrice(): any {
```
Comment (alimoncul) [inline scripts/components/FlProductItem.ts:45]: typings should be set

File: scripts/components/FlRow2LineButton.ts [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    set bottomRightButtonText(value: string) {
+        this.btnBottomRight.text = value;
+    }
+    get mainOnClick(): () => void {
+        return this.__mainOnClick;
+    }
+    set mainOnClick(value: () => void) {
+        this.__mainOnClick = value;
+        this.btnMain.onPress = value;
```
Comment (alimoncul) [inline scripts/components/FlRow2LineButton.ts]: you should use event emitter events

File: scripts/components/GviHomeCategoryItem.ts [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+
+export default class GviHomeCategoryItem extends GviHomeCategoryItemDesign {
+    pageName?: string | undefined;
+    constructor(props?: any, pageName?: string) {
+        // Initalizes super class for this scope
+        super(props);
+        this.pageName = pageName;
+    }
+    get categoryImage(): any {
+        return this.imgLeft.image;
```
Comment (alimoncul) [inline scripts/components/GviHomeCategoryItem.ts]: you dont have to return image instance just return the string assigned

File: scripts/components/GviHomeCategoryItem.ts [hunk 24]
```diff
@@ -0,0 +1,23 @@
...
+    }
+    set categoryImage(value: any) {
+        this.imgLeft.image = value;
```
Comment (alimoncul) [inline scripts/components/GviHomeCategoryItem.ts]: add typings

File: scripts/components/GviProductItem.ts [hunk 25]
```diff
@@ -3,68 +3,69 @@ import Button from '@smartface/native/ui/button';
...
+        this.pageName = pageName;
+    }
+    initIndicator() {
+        this.flProductItem.initIndicator();
+    }
+    toggleIndicator(toggle: boolean): void {
+        this.flProductItem.toggleIndicator(toggle);
+    }
+    get onActionClick(): Button['onTouch'] {
+        return this.flProductItem.btnAddToBasket.onTouch;
```
Comment (alimoncul) [inline scripts/components/GviProductItem.ts]: use event emitter events

File: scripts/components/LviEmptyItem.ts [hunk 26]
```diff
@@ -0,0 +1,31 @@
...
+        // Initalizes super class for this scope
+        super(props);
+        this.pageName = pageName;
+    }
+
+    static getHeight(): number {
+        return getCombinedStyle(`.lviEmptyItem`).height || 0;
+    }
+    static getScreenHeight(): number {
+        return Screen.height / 2;
```
Comment (alimoncul) [inline scripts/components/LviEmptyItem.ts]: definition and function doesn't make sense

File: scripts/components/LviFavorites.ts [hunk 27]
```diff
@@ -17,24 +18,31 @@ export default class LviFavorites extends LviFavoritesDesign {
...
     set itemImage(value: any) {
-        this.imgFavoriteItem.image = Image.createFromFile(`images://${value}`)
+        this.imgFavoriteItem.image = Image.createFromFile(`images://${value}`);
     }
     get itemDesc(): string {
         return this.lblFavroiteItemDesc.text;
     }
     set itemDesc(value: string) {
-        this.lblFavroiteItemDesc.text = value
+        this.lblFavroiteItemDesc.text = value;
```
Comment (alimoncul) [inline scripts/components/LviFavorites.ts]: Favroite
ItemDesc

we shouldn't be shortening names

File: scripts/components/LviHomeCategories.ts [hunk 28]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    static getHeight(): number {
+        return height;
+    }
+    get items(): any[] {
+        return this.__items;
+    }
+    set items(value: any[]) {
+        this.__items = value;
+        this.initGridView();
```
Comment (alimoncul) [inline scripts/components/LviHomeCategories.ts:21]: we shouldn't be setting onItemBind everytime items get set. We need a initial function that sets binding probably inside constructor.

File: scripts/components/LviHomeProducts.ts [hunk 29]
```diff
@@ -1,27 +1,64 @@
...
+    }
+    static getHeight(): number {
+        return originalHeight;
+    }
+    get items(): any[] {
+        return this.__items;
+    }
+    set items(value: any[]) {
+        this.__items = value;
+        this.initGridView();
```
Comment (alimoncul) [inline scripts/components/LviHomeProducts.ts:24]: we shouldn't be setting onItemBind everytime items get set. We need a initial function that sets binding probably inside constructor.

File: .ui/library/gviHomeCategoryItem.cpx (hunk 3 above)
Comment (erenkan) [reply .ui/library/gviHomeCategoryItem.cpx]: done.

File: .ui/library/flDialog.cpx (hunk 2 above)
Comment (erenkan) [reply .ui/library/flDialog.cpx]: done.

File: .ui/library/lviHomeCategories.cpx (hunk 5 above)
Comment (erenkan) [reply .ui/library/lviHomeCategories.cpx]: done.

File: .ui/library/lviHomeCategories.cpx (hunk 6 above)
Comment (erenkan) [reply .ui/library/lviHomeCategories.cpx:35]: done.

File: .ui/library/lviHomeProducts.cpx (hunk 7 above)
Comment (erenkan) [reply .ui/library/lviHomeProducts.cpx]: done.

File: .ui/library/lviProfile.cpx (hunk 8 above)
Comment (erenkan) [reply .ui/library/lviProfile.cpx:25]: done.

File: .ui/library/lviShowcaseHeader.cpx (hunk 9 above)
Comment (erenkan) [reply .ui/library/lviShowcaseHeader.cpx:21]: done.

File: .ui/library/flAlert.cpx (hunk 1 above)
Comment (fuatguzel) [reply .ui/library/flAlert.cpx:1]: done.

File: .ui/library/lviSpacer.cpx (hunk 10 above)
Comment (erenkan) [reply .ui/library/lviSpacer.cpx]: done.

File: .ui/pgAccount.pgx (hunk 11 above)
Comment (erenkan) [reply .ui/pgAccount.pgx]: done.

File: .ui/pgCart.pgx (hunk 12 above)
Comment (erenkan) [reply .ui/pgCart.pgx]: done.

File: .ui/pgCategoryDetail.pgx (hunk 14 above)
Comment (fuatguzel) [reply .ui/pgCategoryDetail.pgx]: done.

File: .ui/pgCategories.pgx (hunk 13 above)
Comment (erenkan) [reply .ui/pgCategories.pgx]: done.

File: .ui/pgCategoryDetail.pgx (hunk 14 above)
Comment (erenkan) [reply .ui/pgCategoryDetail.pgx]: done.

File: .ui/pgFavorites.pgx (hunk 15 above)
Comment (erenkan) [reply .ui/pgFavorites.pgx]: done.

File: .ui/pgHome.pgx (hunk 16 above)
Comment (erenkan) [reply .ui/pgHome.pgx]: done.

File: .ui/pgHomeSlider.pgx (hunk 17 above)
Comment (erenkan) [reply .ui/pgHomeSlider.pgx]: done.

File: scripts/components/FlProductItem.ts (hunk 19 above)
Comment (erenkan) [reply scripts/components/FlProductItem.ts]: done.

File: scripts/components/FlProductItem.ts (hunk 18 above)
Comment (fuatguzel) [reply scripts/components/FlProductItem.ts]: done.

File: scripts/components/LviEmptyItem.ts (hunk 26 above)
Comment (fuatguzel) [reply scripts/components/LviEmptyItem.ts]: done.

File: scripts/components/LviFavorites.ts (hunk 27 above)
Comment (fuatguzel) [reply scripts/components/LviFavorites.ts]: Done.

File: scripts/components/GviHomeCategoryItem.ts (hunk 24 above)
Comment (fuatguzel) [reply scripts/components/GviHomeCategoryItem.ts]: done.

File: scripts/components/GviHomeCategoryItem.ts (hunk 23 above)
Comment (fuatguzel) [reply scripts/components/GviHomeCategoryItem.ts]: done.

File: scripts/components/FlProductItem.ts (hunk 20 above)
Comment (erenkan) [reply scripts/components/FlProductItem.ts]: done.

File: scripts/components/FlRow2LineButton.ts (hunk 22 above)
Comment (erenkan) [reply scripts/components/FlRow2LineButton.ts]: done.

File: scripts/components/GviProductItem.ts (hunk 25 above)
Comment (erenkan) [reply scripts/components/GviProductItem.ts]: done.

**Pull request merged**